File: Solutions/14- Longest Common Prefix/Solution.py
class Solution:
    def longestCommonPrefix(self, strs: [str]) -> str:
        if len(strs) == 1:
            return strs[0]

        if not strs[0]:
            return ""

        for i in range(0,len(strs[0])):
            for j in range(1, len(strs)):
                if not strs[j]:
                    return ""

                if i >= len(strs[j]) or strs[j][i] != strs[0][i]:
                    return strs[0][:i]


        return strs[0]

File: Solutions/14- Longest Common Prefix/Solution.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 1)
            return strs[0];
        if (strs[0].isEmpty())
            return "";

        for (int i = 0 ; i < strs[0].length(); i++) {
            for (int j = 1; j < strs.length; j++) {
                if (strs[j].isEmpty())
                    return "";
                if (i >= strs[j].length() || strs[j].charAt(i) != strs[0].charAt(i)) {
                    return strs[0].substring(0,i);
                }
            }
        }

        return strs[0];
    }

}
